scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingTaskComparator.java

```diff
@@ -53,7 +62,28 @@
      */
     public SchedulingTaskComparator(InternalTask task, InternalJob job) {
         this.task = task;
-        this.ssHashCode = SelectionScript.hashCodeFromList(task.getSelectionScripts());
+
+        if (task.getSelectionScripts() != null) {
+            List<SelectionScript> scriptList = SchedulingMethodImpl.resolveScriptVariables(task.getSelectionScripts(),
+                                                                                           task.getRuntimeVariables());
+
+            for (SelectionScript script : scriptList) {
+                SelectionScript modifiedScript = script;
+                try {
+                    Map<String, Serializable> bindings = SchedulingMethodImpl.createBindingsForSelectionScripts(job,
+                                                                                                                task);
+                    modifiedScript = SchedulingMethodImpl.replaceBindingsInsideScript(script, bindings);
+
+                } catch (Exception e) {
+                    logger.error("", e);
```

Comment:
can we add more info in this error log?

Reply:
like the name of the task/job so it'd be easier to locate what selection script failed from the studio.